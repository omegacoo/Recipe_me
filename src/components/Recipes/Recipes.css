.Recipes {
    text-align: center;
    margin-left: auto;
    margin-right: auto;
    box-shadow: 0 -5px 10px -5px;
    width: 100vw;
    height: 100vh;
}

.Recipes_title {
    margin: 4vh;
    margin-bottom: 2vh;
    text-align: center;
    font-size: 10vw;
}

.RecipesList_container {
    background-image: url('../../assets/open_book.png');
    background-size: 100% 100%;
    background-repeat: no-repeat;
    margin-left: 4%;
    margin-right: auto;
    display: block;
    width: 90vw;
    height: 55vh;
    padding: 2vh;
}

.RecipesList {
    width: 80%;
    margin-left: 15%;
    margin-right: auto;
    margin-bottom: 1vh;
    height: 96%;
    overflow: auto;
    padding: 1vh;
    font-weight: bolder;
    font-size: 2vh;
    border-radius: 15px;
    background-color: #F0EAD6;
}

.Recipes .ingredients {
    background-image: url('../../assets/boxed_food.png');
    background-color: transparent;
    background-repeat: no-repeat;
    background-size: 100%;
    width: 30vw;
    height: 18vh;
    border: none;
    display: block;
    margin-left: auto;
    margin-right: auto;
    font-weight: bolder;
}

/* iPad 768x1024 Portrait */
@media only screen and (min-width: 768px) and (min-height: 1024px) {
    .Recipes_title {
        font-size: 8vw;
        margin-bottom: 0;
    }

    .RecipesList_container {
        width: 80vw;
        margin-left: 8%;
    }

    .Recipes .ingredients {
        width: 20vw;
    }
}

/* Desktop */
@media screen and (min-width: 1025px) {
    .Recipes_title {
        font-size: 5vw;
    }

    .RecipesList_container {
        width: 30vw;
        height: 50vh;
        margin-left: auto;
        margin-right: auto;
    }

    .Recipes .ingredients {
        width: 8vw;
        height: 15vh;
        background-size: 100% 100%;
        margin-top: 2vh;
    }

    .Recipes .ingredients:hover {
        cursor: pointer;
    }
    
    .Recipes .ingredients:active {
        width: 7vw;
        height: 14vh;
        margin-top: 3vh;
    }
    
}