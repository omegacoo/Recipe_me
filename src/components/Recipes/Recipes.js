import React from 'react';
import { Link } from 'react-router-dom';

import RecipeCard from '../RecipeCard/RecipeCard';

import './Recipes.css';

import StoreContext from '../../StoreContext';

export default class Recipes extends React.Component {
    static contextType = StoreContext;

    renderRecipes = () => {
        if(this.context.availableRecipes.length > 0){
            return this.context.availableRecipes.map(r =>
                <RecipeCard 
                    recipe={r}
                    key={r.id}
                />
            );
        } else {
            return 'Sorry, no recipes found...'
        };
    };

    render(){
        return(
            <div className='Recipes'>
                <ul className='RecipesList'>
                    {this.renderRecipes()}
                </ul>
                <Link 
                    className='toIngredients'
                    to={'/ingredients'}
                >
                    Ingredients
                </Link>
            </div>
        );
    };
};