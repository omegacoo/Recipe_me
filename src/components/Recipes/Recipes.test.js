import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Recipes from './Recipes';

describe('Recipes component', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');

        ReactDOM.render(
            <BrowserRouter>
                <Recipes />
            </BrowserRouter>, 
            div
        );
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders UI correctly', () => {
        expect(
        renderer.create(
            <BrowserRouter>
                <Recipes />
            </BrowserRouter>).toJSON()
        ).toMatchSnapshot();
    });
});