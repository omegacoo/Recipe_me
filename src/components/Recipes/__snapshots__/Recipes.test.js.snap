// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recipes component renders UI correctly 1`] = `
<div
  className="Recipes"
>
  <ul
    className="RecipesList"
  >
    Sorry, no recipes found...
  </ul>
  <a
    className="toIngredients"
    href="/ingredients"
    onClick={[Function]}
  >
    Ingredients
  </a>
</div>
`;
